sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala

```diff
@@ -161,6 +163,24 @@ case class ScriptTransformationExec(
 
               if (scriptOutputReader != null) {
                 if (scriptOutputReader.next(scriptOutputWritable) <= 0) {
+                  // There can be a lag between reader read EOF and the process termination.
+                  // If the script fails to startup, this kind of error may be missed.
+                  // So explicitly waiting for the process termination.
+                  val timeout = SQLConf.get.getConf(SQLConf.TRANSFORMATION_EXIT_TIMEOUT)
+                  try {
+                    if (timeout < 0) {
+                      proc.waitFor()
```

Comment:
we should mention it in the config doc that, negative timeout means infinite.

Reply:
doc is fixed and resubmitted.

Reply:
I think we can add `checkValue(timeout > 0)` to the conf and simplify the code here.